lib/kube/proxy/forwarder.go

```diff
@@ -1575,39 +1547,54 @@ func (f *Forwarder) newClusterSessionDirect(ctx authContext, endpoints []kubeClu
 	}
 
 	f.log.WithField("kube_service.endpoints", endpoints).Debugf("Kubernetes session for %v forwarded to remote kubernetes_service instance.", ctx)
-	sess := &clusterSession{
+	return &clusterSession{
 		parent:               f,
 		authContext:          ctx,
 		kubeClusterEndpoints: endpoints,
 		tlsConfig:            tlsConfig,
 		// This session talks to a kubernetes_service, which should handle
 		// audit logging. Avoid duplicate logging.
 		noAuditEvents: true,
+	}, nil
+}
+
+func (f *Forwarder) setSessionForwarder(sess *clusterSession) error {
```

Comment:
nit: I like to see short godoc comments on all functions. It would help clarify their purpose and help new developers learn the codebase.